cloudformation/external-slurmdbd/external_slurmdbd/external_slurmdbd_stack.py

```diff
@@ -61,42 +82,49 @@ def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
         # create management security group with SSH access from anywhere (TEMPORARY!)
         self._ssh_server_sg, self._ssh_client_sg = self._add_management_security_groups()
 
-        # create Launch Template
-        self._launch_template = self._add_external_slurmdbd_launch_template()
-
-        # define EC2 Auto Scaling Group (ASG)
-        self._external_slurmdbd_asg = self._add_external_slurmdbd_auto_scaling_group()
+        # create a pair of security groups for the slurm accounting traffic across
+        # between cluster head node and external slurmdbd instance via port 6819
+        self._slurmdbd_server_sg, self._slurmdbd_client_sg = self._add_slurmdbd_accounting_security_groups()
 
         # Create a CloudWatch log group
         self._log_group = self._add_cloudwatch_log_group()
 
         # define IAM role and necessary IAM policies
         self._role = self._add_iam_role()
 
+        self._instance_profile = self._add_instance_profile(self._role.ref, "ExternalSlurmdbdInstanceProfile")
+
+        # create Launch Template
+        self._launch_template = self._add_external_slurmdbd_launch_template()
+
+        # define EC2 Auto Scaling Group (ASG)
+        self._external_slurmdbd_asg = self._add_external_slurmdbd_auto_scaling_group()
+
     def _add_cfn_init_config(self):
         dna_json_content = {
-            "dbms_uri": self.dbms_uri_param.value_as_string,
-            "dbms_username": self.dbms_username_param.value_as_string,
-            "dbms_database_name": self.dbms_database_name_para.value_as_string,
-            "dbms_password_secret_arn": self.dbms_password_secret_arn_param.value_as_string,
-            "munge_key_secret_arn": self.munge_key_secret_arn_param.value_as_string,
-            "region": self.stack.region,
-            "stack_name": self.stack.stack_name,
+            "dbms_uri": self.dbms_uri.value_as_string,
+            "dbms_username": self.dbms_username.value_as_string,
+            "dbms_database_name": self.dbms_database_name.value_as_string,
+            "dbms_password_secret_arn": self.dbms_password_secret_arn.value_as_string,
+            "munge_key_secret_arn": self.munge_key_secret_arn.value_as_string,
+            "region": self.region,
+            "stack_name": self.stack_name,
             "nlb_dns_name": self._external_slurmdbd_nlb.load_balancer_dns_name,
-            "if_external_slurmdbd": True,
+            "is_external_slurmdbd": True,
         }
 
         return {
```

Comment:
Maybe we can add this as part of the Write_files section in user_data.sh

Reply:
Is there any benefit of using "Write_files". If not, we can keep the code as is

Reply:
Agreed. Infact I see that HN, CF and keep the dna_json data separate from user_data.sh